ansible/roles/prosody/files/mod_muc_webhooks.lua

```diff
@@ -802,20 +809,31 @@ local function occupant_affiliation_changed(event)
     if event.actor and event.affiliation == 'owner' then
         local room = event.room;
         local granted_to_occupant = {};
+        local granted_by_occupant = {};
         -- event.jid is the bare jid of participant
         for _, occupant in room:each_occupant() do
             if occupant.bare_jid == event.jid then
                 granted_to_occupant = occupant;
-                module:log("debug", "Participant %s was promoted to moderator by %s", occupant.jid, event.actor)
+                module:log("debug", "Participant %s was promoted to moderator by %s", occupant.jid, event.actor);
+            end
+            if occupant.jid == event.actor then
+                granted_by_occupant = occupant;
+            end
+            -- break when both found
+            if next(granted_to_occupant) and next(granted_by_occupant) then
+                break;
             end
         end
-
         local eventData = {
             grantedBy = {
-                participantJid = event.actor
+                participantJid = granted_by_occupant.jid,
```

Comment:
@damencho is there a reason why these are full jids, usually I saw that we use the bare_jid for webhooks